### PR TITLE
Implement genericblock/generichide filter options

### PR DESCRIPTION
**Pull request by tomasko126**
@kpeckett This PR adds suppport for genericblock/generichide support.

Please review and test on these following sites:
https://static.kzar.co.uk/genericblockhide-test/index.html
https://holly.cat/genericblockhide-test.html - loads the first URL in a frame

Thanks!

### CODE REVIEW

**Comment by tomasko126** [comment]
This PR fixes #95.

**Comment by tomasko126** [comment]
@kpeckett I haven't implemented support for these filter options for CatBlock on Safari. This will come likely in another release.

File: filtering/filteroptions.js [hunk 1]
```diff
@@ -14,7 +14,9 @@ var ElementTypes = {
     xmlhttprequest: 512,
     "document": 1024,
     elemhide: 2048,
-    popup: 4096
+    popup: 4096,
+    genericblock: 8192,
+    generichide: 16384
     // If you add something here, update .DEFAULTTYPES and .CHROMEONLY below.
```
Comment (itskdog) [inline filtering/filteroptions.js:20]: Did you do what this comment asks?

**Comment by itskdog** [comment]
What should I be looking for in the test link?

**Comment by tomasko126** [comment]
@kpeckett 

If you add the following filter: `@@||holly.cat$genericblock,generichide`, you shouldn't see anything blocked on https://holly.cat/genericblockhide-test.html.

If you add the following filter: `@@||static.kzar.co.uk$genericblock,generichide`, you shouldn't see anything blocked on https://holly.cat/genericblockhide-test.html & on https://static.kzar.co.uk/genericblockhide-test/index.html as well.

Checkout https://adblockplus.org/filters#options for more info about genericblock & generichide filter options.

File: filtering/filteroptions.js (hunk 1 above)
Comment (tomasko126) [reply filtering/filteroptions.js:20]: @kpeckett Good catch. I will add support for Safari as well.

**Comment by tomasko126** [comment]
We won't implement new features like this one for Safari, since the API we are using for blocking ads is now obsolete.

**Pull request merged**